bleep-core/src/scala/bleep/GenBloopFiles.scala

```diff
@@ -328,6 +331,52 @@ object GenBloopFiles {
         case _                                            => bloop.config.Tag.Library
       }
 
+    val sourceGenerators = {
+      val sg = explodedProject.`source-generator-scripts`.values.map { scriptName =>
+        if (!build.scripts.contains(scriptName)) {
```

Comment:
style:

```scala
val script = build.scripts.get(scriptName).getOrElse(
  throw new BleepException.Text(crossName, s"Unknown script $scriptName")
)
```

Avoid looking up `build.scripts.get(scriptName)` further down.

Difference between `sys.error` and `BleepException` in bleep is that the latter will not show a stack trace to the user (only under `--debug` or in the log file)